Network/Curl/Easy.hs

```diff
@@ -112,6 +112,12 @@ setopt hh o = curlPrim hh $ \ r h -> unmarshallOption (easy_um r h) o
             fp <- mkWriter x
             updateCleanup r i $ debug "FREE: WRITER" >> freeHaskellFunPtr fp
             liftM toCode $ easy_setopt_wfun h i fp
+     , u_writeFunH -- :: Int -> WriteFunction -> IO a
+       = \ i x -> do
+            debug "ALLOC: HEADER WRITER"
+            fp <- mkWriter x
+            updateCleanup r i $ debug "FREE: HEADER WRITER" >> freeHaskellFunPtr fp
```

Comment:
We use the `freeHaskellFunPtf fp`, but still use `fp` after that. Is that safe?

Reply:
I don't quite understand, where do we use `fp` after `freeHaskellFunPtf fp`?

Reply:
In the final line, we have `liftM toCode $ easy_setopt_wfun h i fp`, so we use `fp` after that. On the other hand, all other places use it as well, so it is probably fine.

Reply:
Yeah, the existing code is pretty obscurely written in that `updateCleanup` adds an `IO ()` action to be run when `runCleanup` is called. So the `fp` is still allocated when `liftM` happens but is deallocated when the `Curl` handle is garbage collected or `reset`.

Reply:
Ah, that makes sense, thanks for the explanation.